### DISCUSSION TITLE
Send output to Web API with Elsa 2.13

### DISCUSSION THREAD

**Discussion by Rathna314** [elsa-workflows/elsa-core#5500] [Q&A]
How do I send output of a custom activity back to WebAPI from Elsa workflow. I am using version 2.13?

**Comment by glime-ai** [comment]
To send the output of a custom activity back to a WebAPI in Elsa workflow version 2.13, you would typically use the `WriteHttpResponse` activity to return a response from your workflow to the calling client. Here's a general approach on how to do this:

1. Define your custom activity with an output property.
2. In your workflow, after executing the custom activity, capture its output.
3. Use the `WriteHttpResponse` activity to send the captured output back as an HTTP response.

Here's an example of how you might define a custom activity with an output:

```csharp
public class MyCustomActivity : Activity
{
    public Output<string> MyOutput { get; set; }

    protected override IActivityExecutionResult OnExecute(ActivityExecutionContext context)
    {
        // Your custom logic here
        string result = "Some result";
        
        // Set the output
        MyOutput.Set(context, result);

        return Done();
    }
}
```

And then in your workflow:

```csharp
builder
    .StartWith<MyCustomActivity>()
    .Then<WriteHttpResponse>(setup =>
    {
        setup.Content = new JavaScriptExpression<string>("MyCustomActivityOutput");
        setup.ContentType = "application/json";
        setup.StatusCode = HttpStatusCode.OK;
    });
```

In this example, `MyCustomActivityOutput` would be the variable where you store the output of `MyCustomActivity`. You would need to ensure that you have correctly set up the output binding in your workflow definition.

Please note that the exact syntax and method to achieve this may vary depending on the specifics of your custom activity and workflow setup. If you need more detailed guidance, you should refer to the Elsa documentation or seek help from the Elsa community, as the provided context does not include specific details about creating custom activities and handling their outputs in version 2.13.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by Rathna314** [reply]
I don't find public Output<string> reference with Elsa 2.13
